### PR TITLE
Apb 700 skeleton

### PR DESCRIPTION
**Pull request by PaulHodgson**

### CODE REVIEW

File: app/uk/gov/hmrc/agentmappingfrontend/controllers/MappingController.scala [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+import uk.gov.hmrc.play.frontend.auth.connectors.AuthConnector
+import uk.gov.hmrc.play.frontend.controller.FrontendController
+
+import scala.concurrent.Future.successful
+
+@Singleton
+class MappingController @Inject()(override val messagesApi: MessagesApi, override val authConnector: AuthConnector)  (implicit appConfig: AppConfig)
+  extends FrontendController with I18nSupport with AuthActions {
+  val start: Action[AnyContent] = Action { implicit request =>
+    Ok(html.start_template.apply())
```
Comment (bravesirrobin81) [inline app/uk/gov/hmrc/agentmappingfrontend/controllers/MappingController.scala]: I think you can elide the `.apply()` to `()`

File: it/uk/gov/hmrc/agentmappingfrontend/controllers/MappingControllerISpec.scala [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    "display the start page" in {
+      val result: Result = await(controller.start(FakeRequest()))
+      status(result) shouldBe 200
+      bodyOf(result) should include("map existing agent codes to your new MTD Subscription")
+    }
+  }
+
+  "add-code" should {
+
+    behave like anEndfpointAccessableGivenAgentAffinityGroupAndEnrolmentIrPayeAgent(request => controller.addCode(request))
```
Comment (bravesirrobin81) [inline it/uk/gov/hmrc/agentmappingfrontend/controllers/MappingControllerISpec.scala]: These should refer to  IR SA Agent, I think.

Comment (bravesirrobin81) [inline it/uk/gov/hmrc/agentmappingfrontend/controllers/MappingControllerISpec.scala]: Also, 'Endfpoint'

File: app/uk/gov/hmrc/agentmappingfrontend/GuiceModule.scala [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+  }
+
+  private def bindBaseUrl(serviceName: String) =
+    bind(classOf[URL]).annotatedWith(named(s"$serviceName-baseUrl")).toProvider(new BaseUrlProvider(serviceName))
+
+  private class BaseUrlProvider(serviceName: String) extends Provider[URL] {
+    override lazy val get = new URL(baseUrl(serviceName))
+  }
+
+  private def bindConfigProperty(propertyName: String) =
```
Comment (PaulHodgson) [inline app/uk/gov/hmrc/agentmappingfrontend/GuiceModule.scala]: good spot. removed.

File: app/uk/gov/hmrc/agentmappingfrontend/auth/AuthActions.scala [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+import scala.concurrent.Future
+
+case class AgentRequest[A](enrolments: List[Enrolment], request: Request[A]) extends WrappedRequest[A](request)
+
+trait AuthActions extends Actions {
+  protected type AsyncPlayUserRequest = AuthContext => AgentRequest[AnyContent] => Future[Result]
+  protected type PlayUserRequest = AuthContext => AgentRequest[AnyContent] => Result
+  private implicit def hc(implicit request: Request[_]): HeaderCarrier = fromHeadersAndSession(request.headers, Some(request.session))
+
+  def AuthorisedWithSubscribingAgentAsync(body: AsyncPlayUserRequest): Action[AnyContent] =
```
Comment (PaulHodgson) [inline app/uk/gov/hmrc/agentmappingfrontend/auth/AuthActions.scala]: Done.

File: app/uk/gov/hmrc/agentmappingfrontend/auth/AuthActions.scala [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+    AuthorisedFor(NoOpRegime, pageVisibility = GGConfidence).async {
+      implicit authContext => implicit request =>
+        isAgentAffinityGroup() flatMap {
+          case true => enrolments flatMap {
+              e => {
+                if ( e.exists(_.key == "IR-PAYE-AGENT") ){
```
Comment (PaulHodgson) [inline app/uk/gov/hmrc/agentmappingfrontend/auth/AuthActions.scala]: Whoops. Also fixed the corresponding test data.

File: it/uk/gov/hmrc/agentmappingfrontend/controllers/MappingControllerISpec.scala (hunk 2 above)
Comment (PaulHodgson) [inline it/uk/gov/hmrc/agentmappingfrontend/controllers/MappingControllerISpec.scala]: Done

File: app/uk/gov/hmrc/agentmappingfrontend/controllers/MappingController.scala (hunk 1 above)
Comment (PaulHodgson) [inline app/uk/gov/hmrc/agentmappingfrontend/controllers/MappingController.scala]: Done.

File: app/uk/gov/hmrc/agentmappingfrontend/config/frontendAppConfig.scala [hunk 6]
```diff
@@ -10,15 +29,34 @@ trait AppConfig {
   val reportAProblemNonJSUrl: String
 }
 
-object FrontendAppConfig extends AppConfig with ServicesConfig {
+trait StrictConfig{
+  def loadConfig(key: String): String = configuration.getString(key).getOrElse(throw new Exception(s"Missing configuration key: $key"))
+}
 
-  private def loadConfig(key: String) = configuration.getString(key).getOrElse(throw new Exception(s"Missing configuration key: $key"))
+object GGConfig extends StrictConfig {
```
Comment (PaulHodgson) [inline app/uk/gov/hmrc/agentmappingfrontend/config/frontendAppConfig.scala]: Consistency is good but in this case the config id needed by the AuthActions trait, and the price of making its injectable is additional parameters elsewhere.

**Pull request merged**